Zend/zend_API.c

```diff
@@ -4505,6 +4505,19 @@ ZEND_API zend_property_info *zend_declare_typed_property(zend_class_entry *ce, z
 	if (!(access_type & ZEND_ACC_PPP_MASK)) {
 		access_type |= ZEND_ACC_PUBLIC;
 	}
+	if (access_type & ZEND_ACC_PRIVATE_SET) {
+		access_type |= ZEND_ACC_FINAL;
+	}
```

Comment:
Shouldn't this be moved 10 lines below, until equivalent checks are removed (i.e. `private private(set)` should not cause final being implied)?